development-workflow.md

```diff
@@ -0,0 +1,61 @@
+# MeiliSearch Development Workflow
```

Comment:
Should this be in the main repo than? And the README in this repo pointing there?

Reply:
I was wondering where to put this document. This is not a contributing guide but more of a description of how we do things today.

Should we update the contributing guide on main repo and link this workflow description here? (not sure about that)

Reply:
Currently, the README is explaining in a few words what each part of the specs file should be. This document is nice to have to explain how we operate at a higher level. Should we mention it from the README and also on the main repo to lead people here?